x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_create_extension.tsx

```diff
@@ -5,45 +5,535 @@
  * for use in the Ingest app create / edit package policy
  */
+
+const ENDPOINT_INTEGRATION_CONFIG_KEY = 'ENDPOINT_INTEGRATION_CONFIG';
+
+const NGAV = 'NGAV';
+const EDR_ESSENTIAL = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.endpointDropdownOptionEDREssential',
+  {
+    defaultMessage: 'EDR Essential',
+  }
+);
+const EDR_COMPLETE = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.endpointDropdownOptionEDRComplete',
+  {
+    defaultMessage: 'EDR Complete',
+  }
+);
+
+const ENDPOINT = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.endpointDropdownOption',
+  {
+    defaultMessage: 'Endpoint',
+  }
+);
+const CLOUD_SECURITY = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.cloudSecurityDropdownOption',
+  {
+    defaultMessage: 'Cloud Security',
+  }
+);
+const INTERACTIVE_ONLY = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.cloudEventFiltersInteractiveOnly',
+  {
+    defaultMessage: 'Interactive only',
+  }
+);
+const ALL_EVENTS = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.cloudEventFiltersAllEvents',
+  {
+    defaultMessage: 'All events',
+  }
+);
+const PREVENT_MALWARE = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.cloudEventFiltersPreventionMalware',
+  {
+    defaultMessage: 'Prevent Malware',
+  }
+);
+const PREVENT_MALICIOUS_BEHAVIOUR = i18n.translate(
+  'xpack.securitySolution.createPackagePolicy.stepConfigure.cloudEventFiltersPreventionMaliciousBehaviour',
+  {
+    defaultMessage: 'Prevent Malicious Behaviour',
+  }
+);
+
+const SelectConfigurationSettingsText = styled.div`
+  font-size: 22px;
+  font-weight: 700;
+  padding-top: 12px;
+  padding-bottom: 10px;
+  line-height: 32px;
+`;
+
+const RadioOptionsDetails = styled.div`
+  padding-left: 24px;
+  color: ${(props) => props.theme.eui.euiColorDarkShade};
+  font-weight: 400;
+  font-size: 12.25px;
+  line-height: 21px;
+`;
+
+const IntegrationOptionsContainer = styled.div`
+  margin-bottom: 48px;
+  margin-top: 17px;
+`;
+
+const CloudOptionDataIngestionMessage = styled.div`
+  font-size: 14px;
+  width: max-content;
+  font-weight: 400;
+  padding-top: 12px;
+  padding-bottom: 10px;
+  color: ${(props) => props.theme.eui.euiColorDarkShade};
+`;
+
+const SubduedText = styled.div`
+  font-size: 14px;
+  font-weight: 400;
+  color: ${(props) => props.theme.eui.euiColorDarkShade};
+  line-height: 24px;
+`;
+
+const BoldSubtitle = styled.div`
+  font-size: 14px;
+  font-weight: 700;
+  padding-top: 10px;
+  padding-bottom: 8px;
+`;
+
+const QuickSettingInfo = styled.div`
+  padding-bottom: 16px;
+  font-size: 14px;
+  line-height: 24px;
+`;
+
+const CloudRadioProtectionsModeContainer = styled.div`
+  padding-top: 8px;
+`;
+
+const PaddingBottomText = styled.div`
+  padding-bottom: 14px;
+`;
+
+const dropDownOptions = [
```

Comment:
can you make this read only? or use `Object.freeze() on the array.

also, can you add a type to it.

Reply:
@paul-tavares - Thanks for all the tips on the PR review, I pushed a code refactor that addresses most of the changes, so I'll ask you to review one more time, but I'm unsure if this one is still needed now that I added the mappings.